// @ssgoi/react@4.2.1/dist/core/dist/types.js
function p(e) {
  return ("tick" in e || "css" in e) && !("items" in e);
}
function u(e) {
  return "items" in e;
}
function d(e) {
  return "css" in e && typeof e.css == "function";
}
function h(e) {
  return "tick" in e && typeof e.tick == "function";
}
async function k(e) {
  const n = await e;
  return u(n) ? {
    prepare: n.prepare,
    wait: n.wait,
    onStart: n.onStart,
    onEnd: n.onEnd,
    items: n.items,
    schedule: n.schedule
  } : {
    prepare: n.prepare,
    wait: n.wait,
    onStart: n.onStart,
    onEnd: n.onEnd,
    items: [
      {
        physics: n.physics,
        tick: n.tick,
        css: n.css
      }
    ],
    schedule: "parallel"
  };
}
function y(e, n) {
  const a = e.schedule ?? "parallel", c = e.items.map(
    (s) => {
      let t;
      switch (a) {
        case "parallel":
          t = 0;
          break;
        case "sequential":
          t = 1;
          break;
        case "stagger":
          t = Math.max(0, Math.min(1, s.offset ?? 0));
          break;
        default:
          t = 0;
      }
      const { css: i, offset: f, ...r } = s, l = i ? typeof i == "function" ? n ? { element: n, style: i } : void 0 : { element: i.element, style: i.style } : void 0;
      return { ...r, css: l, normalizedOffset: t };
    }
  ), { schedule: m, ...o } = e;
  return {
    ...o,
    items: c
  };
}
export {
  d as isCssAnimation,
  u as isMultiAnimation,
  p as isSingleAnimation,
  h as isTickAnimation,
  y as normalizeSchedule,
  k as normalizeToMultiAnimation
};
