// @ssgoi/react@4.2.1/dist/react/src/lib/ssgoi-transition.js
"use client";
import { jsx as d } from "react/jsx-runtime";
import { useRef as r, useCallback as C, useMemo as u } from "react";
import { transition as k } from "./transition.js";
import { useSsgoi as x } from "./context.js";
import { forkStyleFromElement as E, combineRefs as F } from "./utils.js";
const h = ({
  children: m,
  id: o,
  as: a,
  className: R,
  ...p
}) => {
  const { getTransition: g, getInitialStyle: S } = x(), y = a || "div", e = r(null), i = r(!0), t = r(null);
  t.current === null && (t.current = S());
  let n;
  i.current ? n = t.current : e.current && t.current && (n = E(
    e.current,
    t.current
  ));
  const s = C((l) => {
    e.current = l, l && (i.current = !1);
  }, []), c = g(o), f = u(
    () => k(c),
    [c]
  ), b = u(
    () => F(s, f),
    [s, f]
  );
  return /* @__PURE__ */ d(
    y,
    {
      ref: b,
      "data-ssgoi-transition": o,
      className: R,
      style: n,
      ...p,
      children: m
    }
  );
};
export {
  h as SsgoiTransition
};
