// @ssgoi/core@4.4.0/dist/index.js
var it = Object.defineProperty;
var nt = (e, n, t) => n in e ? it(e, n, { enumerable: !0, configurable: !0, writable: !0, value: t }) : e[n] = t;
var d = (e, n, t) => nt(e, typeof n != "symbol" ? n + "" : n, t);
import { normalizeToMultiAnimation as M, normalizeSchedule as j } from "./types.js";
import { isCssAnimation as Xt, isMultiAnimation as Yt, isSingleAnimation as Ht, isTickAnimation as jt } from "./types.js";
import { S as B, I as rt, a as ot } from "./provider-Cu4NSF1Q.js";
class st {
  constructor() {
    d(this, "listeners", /* @__PURE__ */ new Set());
    d(this, "rafId", null);
    d(this, "isRunning", !1);
    // Timing
    d(this, "startTime", Date.now());
    d(this, "lastUpdate", this.startTime);
    d(this, "elapsed", 0);
    // Lag smoothing (mobile optimization)
    d(this, "lagThreshold", 500);
    // ms - detect tab switches, backgrounding
    d(this, "adjustedLag", 33);
    // ms - max time jump to prevent animation skip
    d(this, "maxDeltaTime", 33);
    // ms - clamp for physics stability (30fps minimum)
    // FPS throttling
    d(this, "gap", 1e3 / 240);
    // max 240fps
    d(this, "nextTime", this.gap);
    d(this, "tick", () => {
      const n = Date.now();
      let i = n - this.lastUpdate;
      i > this.lagThreshold || i < 0 ? (this.startTime += i - this.adjustedLag, i = this.adjustedLag) : i > this.maxDeltaTime && (i = this.maxDeltaTime), this.lastUpdate = n;
      const r = n - this.startTime, o = r - this.nextTime;
      if (o > 0) {
        const s = i / 1e3;
        this.elapsed = r / 1e3, this.nextTime += o + (o >= this.gap ? 4 : this.gap - o), this.listeners.forEach((c) => {
          c(s, this.elapsed);
        });
      }
      this.listeners.size > 0 ? this.rafId = requestAnimationFrame(this.tick) : (this.isRunning = !1, this.rafId = null);
    });
  }
  subscribe(n) {
    return this.listeners.add(n), this.isRunning || (this.isRunning = !0, this.rafId = requestAnimationFrame(this.tick)), () => {
      this.unsubscribe(n);
    };
  }
  unsubscribe(n) {
    this.listeners.delete(n), this.listeners.size === 0 && this.rafId !== null && (cancelAnimationFrame(this.rafId), this.isRunning = !1, this.rafId = null);
  }
  /**
   * Configure lag smoothing behavior
   * @param threshold - Time threshold to detect lag (default: 500ms)
   * @param adjustedLag - Max time jump when lag detected (default: 33ms)
   */
  lagSmoothing(n, t) {
    this.lagThreshold = n, this.adjustedLag = Math.min(t, n);
  }
  /**
   * Set maximum FPS
   * @param fps - Target FPS (default: 240)
   */
  fps(n) {
    this.gap = 1e3 / n, this.nextTime = this.elapsed * 1e3 + this.gap;
  }
  getListenerCount() {
    return this.listeners.size;
  }
  getElapsed() {
    return this.elapsed;
  }
}
const N = new st();
function at(e) {
  const {
    integrator: n,
    from: t,
    to: i,
    velocity: r = 0,
    onUpdate: o,
    onComplete: s,
    onStart: c
  } = e;
  let a = { position: t, velocity: r }, f = !0, y = 0, C = !1;
  const A = (E) => {
    if (f) {
      if (C || (C = !0, c == null || c()), a = n.step(a, i, E), n.isSettled(a, i)) {
        if (y += Math.min(E, 0.033), y >= B) {
          a = { position: i, velocity: 0 }, f = !1, o(a.position), N.unsubscribe(A), s();
          return;
        }
      } else
        y = 0;
      o(a.position);
    }
  };
  return N.subscribe(A), {
    stop: () => {
      f && (f = !1, N.unsubscribe(A));
    },
    getPosition: () => a.position,
    getVelocity: () => a.velocity,
    isRunning: () => f
  };
}
const O = 1e3 / 60;
function ct(e, n, t, i = 0) {
  let o = { position: n, velocity: i }, s = 0;
  const c = [];
  for (let a = 0; a < 600; a++) {
    const f = a * O;
    if (c.push({
      time: f,
      position: o.position,
      velocity: o.velocity
    }), o = e.step(o, t, O / 1e3), e.isSettled(o, t)) {
      if (s += O / 1e3, s >= B) {
        c.push({
          time: (a + 1) * O,
          position: t,
          velocity: 0
        });
        break;
      }
    } else
      s = 0;
  }
  return c;
}
function W(e, n) {
  if (e.length === 0)
    return { position: 0, velocity: 0 };
  const t = e[0], i = e[e.length - 1];
  if (n <= 0)
    return { position: t.position, velocity: t.velocity };
  if (n >= i.time)
    return { position: i.position, velocity: i.velocity };
  let r = 0, o = e.length - 1;
  for (; r < o - 1; ) {
    const f = Math.floor((r + o) / 2);
    e[f].time <= n ? r = f : o = f;
  }
  const s = e[r], c = e[o], a = (n - s.time) / (c.time - s.time);
  return {
    position: s.position + (c.position - s.position) * a,
    velocity: s.velocity + (c.velocity - s.velocity) * a
  };
}
function lt(e, n) {
  return e.map((t) => n(t.position));
}
function ut(e) {
  const {
    integrator: n,
    element: t,
    from: i,
    to: r,
    velocity: o = 0,
    style: s,
    onComplete: c,
    onStart: a
  } = e, f = ct(n, i, r, o);
  if (f.length === 0)
    return a == null || a(), c(), {
      stop: () => {
      },
      getPosition: () => r,
      getVelocity: () => 0,
      isRunning: () => !1
    };
  const y = lt(f, s), C = f[f.length - 1], A = C.time, E = y[0];
  if (E)
    for (const l of Object.keys(E))
      t.style[l] = "";
  const g = t.animate(y, {
    duration: A,
    fill: "forwards",
    easing: "linear",
    composite: "replace"
  });
  let m = !0;
  const h = performance.now();
  return a == null || a(), g.onfinish = () => {
    m && (m = !1, c());
  }, {
    stop: () => {
      m && (m = !1, g.cancel());
    },
    getPosition: () => {
      if (!m)
        return C.position;
      const l = performance.now() - h;
      return W(f, l).position;
    },
    getVelocity: () => {
      if (!m)
        return 0;
      const l = performance.now() - h;
      return W(f, l).velocity;
    },
    isRunning: () => m
  };
}
function ft(e) {
  const { to: n, onStart: t, onComplete: i } = e;
  return t == null || t(), i(), {
    stop: () => {
    },
    getPosition: () => n,
    getVelocity: () => 0,
    isRunning: () => !1
  };
}
class mt {
  /**
   * Get bound runner based on animation mode
   *
   * @param options Animation mode options (tick or css)
   * @returns Bound runner function (empty runner if no animation mode specified)
   * @throws Error if both tick and css are provided
   */
  static from(n) {
    const { tick: t, css: i } = n;
    if (t && i)
      throw new Error("Cannot use both 'tick' and 'css' options together");
    return i ? (r) => ut({
      ...r,
      element: i.element,
      style: i.style
    }) : t ? (r) => at({
      ...r,
      onUpdate: t
    }) : ft;
  }
}
class J {
}
const dt = { stiffness: 300, damping: 30 };
class G extends J {
  constructor(t) {
    super();
    d(this, "options");
    d(this, "runner");
    d(this, "controls", null);
    d(this, "isAnimating", !1);
    d(this, "currentValue");
    d(this, "currentVelocity", 0);
    d(this, "updateFn");
    if (this.options = {
      from: t.from ?? 0,
      to: t.to ?? 1,
      physics: t.physics,
      onComplete: t.onComplete ?? (() => {
      }),
      onStart: t.onStart
    }, this.currentValue = this.options.from, t.tick)
      this.updateFn = (i) => {
        var r;
        return (r = t.tick) == null ? void 0 : r.call(t, i);
      };
    else if (t.css) {
      const { element: i, style: r } = t.css;
      this.updateFn = (o) => {
        const s = r(o);
        for (const [c, a] of Object.entries(s))
          i.style[c] = typeof a == "number" ? String(a) : a;
      };
    } else
      this.updateFn = () => {
      };
    this.runner = mt.from({
      tick: t.tick,
      css: t.css
    });
  }
  /**
   * Sync element state to current progress value
   * Uses the update callback if provided
   */
  syncState() {
    var t;
    (t = this.updateFn) == null || t.call(this, this.currentValue);
  }
  /**
   * Create Integrator from physics options
   * Priority: integrator factory > inertia/spring config > default spring
   */
  createIntegrator() {
    const t = this.options.physics;
    return t != null && t.integrator ? t.integrator() : t != null && t.spring || t != null && t.inertia ? rt.from({
      spring: t.spring,
      inertia: t.inertia
    }) : new ot(dt);
  }
  runAnimation(t) {
    this.isAnimating = !0, this.controls = this.runner({
      integrator: this.createIntegrator(),
      from: this.currentValue,
      to: t,
      velocity: this.currentVelocity,
      onStart: this.options.onStart,
      onComplete: () => {
        this.currentValue = t, this.currentVelocity = 0, this.isAnimating = !1, this.controls = null, this.options.onComplete();
      }
    });
  }
  forward() {
    this.stop(), this.runAnimation(this.options.to);
  }
  backward() {
    this.stop(), this.runAnimation(this.options.from);
  }
  reverse() {
    const t = this.options.from;
    this.options.from = this.options.to, this.options.to = t, this.isAnimating && (this.stop(), this.runAnimation(this.options.to));
  }
  stop() {
    this.controls && (this.currentValue = this.controls.getPosition(), this.currentVelocity = this.controls.getVelocity(), this.controls.stop(), this.controls = null), this.isAnimating = !1;
  }
  getVelocity() {
    var t;
    return ((t = this.controls) == null ? void 0 : t.getVelocity()) ?? this.currentVelocity;
  }
  getCurrentValue() {
    var t;
    return ((t = this.controls) == null ? void 0 : t.getPosition()) ?? this.currentValue;
  }
  getIsAnimating() {
    return this.isAnimating;
  }
  getCurrentState() {
    return {
      position: this.getCurrentValue(),
      velocity: this.getVelocity(),
      from: this.options.from,
      to: this.options.to
    };
  }
  setValue(t) {
    this.currentValue = t;
  }
  setVelocity(t) {
    this.currentVelocity = t;
  }
  updateOptions(t) {
    this.options = { ...this.options, ...t };
  }
}
class D extends J {
  constructor(t) {
    super();
    d(this, "config");
    d(this, "animators", /* @__PURE__ */ new Map());
    d(this, "itemOrder", []);
    d(this, "completedCount", 0);
    d(this, "completedAnimators", /* @__PURE__ */ new Set());
    d(this, "direction", "forward");
    d(this, "idCounter", 0);
    d(this, "rafId", null);
    d(this, "from");
    d(this, "to");
    d(this, "initialState");
    this.config = t.config, this.from = t.from, this.to = t.to, this.initialState = t.state ?? {
      position: t.from,
      velocity: 0
    }, this.initializeAnimators();
  }
  generateId() {
    return `spring_${this.idCounter++}`;
  }
  initializeAnimators() {
    this.config.items.forEach((t, i) => {
      const r = this.generateId(), o = new G({
        from: this.from,
        to: this.to,
        physics: t.physics,
        tick: t.tick,
        css: t.css,
        onComplete: () => this.onAnimatorComplete(r),
        onStart: t.onStart
      });
      (i === 0 || t.normalizedOffset === 0) && (o.setValue(this.initialState.position), o.setVelocity(this.initialState.velocity), o.syncState()), this.animators.set(r, {
        id: r,
        item: t,
        animator: o,
        started: !1
      }), this.itemOrder.push(r);
    });
  }
  /**
   * Calculate progress of an animator (0-1)
   * Progress = |position - from| / |to - from|
   */
  getAnimatorProgress(t) {
    const i = t.animator.getCurrentState(), r = Math.abs(i.to - i.from);
    return r === 0 ? 1 : Math.abs(i.position - i.from) / r;
  }
  /**
   * Get order of springs based on direction
   * Forward: 0, 1, 2, ...
   * Backward: n-1, n-2, ..., 0
   */
  getOrderedIds() {
    return this.direction === "forward" ? this.itemOrder : [...this.itemOrder].reverse();
  }
  onAnimatorComplete(t) {
    var r, o, s, c, a, f;
    const i = this.animators.get(t);
    !i || this.completedAnimators.has(t) || (this.completedAnimators.add(t), this.completedCount++, (o = (r = i.item).onComplete) == null || o.call(r), (c = (s = this.config).onProgress) == null || c.call(s, this.completedCount, this.config.items.length), this.completedCount === this.config.items.length && (this.stopScheduler(), (f = (a = this.config).onEnd) == null || f.call(a)));
  }
  /**
   * Check and start springs based on previous spring's progress
   * Called on each frame while animation is running
   */
  checkAndStartSprings() {
    const t = this.getOrderedIds();
    for (let i = 0; i < t.length; i++) {
      const r = t[i], o = this.animators.get(r);
      if (!o || o.started) continue;
      if (i === 0) {
        this.startAnimator(r);
        continue;
      }
      const s = t[i - 1], c = this.animators.get(s);
      if (!c || !c.started) continue;
      const a = this.getAnimatorProgress(c), f = o.item.normalizedOffset;
      a >= f && this.startAnimator(r);
    }
  }
  startAnimator(t) {
    const i = this.animators.get(t);
    !i || i.started || (i.started = !0, this.direction === "forward" ? i.animator.forward() : i.animator.backward());
  }
  /**
   * Start the scheduler loop that checks spring progress
   */
  startScheduler() {
    const t = () => {
      this.checkAndStartSprings(), this.getIsAnimating() && (this.rafId = requestAnimationFrame(t));
    };
    t();
  }
  stopScheduler() {
    this.rafId !== null && (cancelAnimationFrame(this.rafId), this.rafId = null);
  }
  getFirstAnimator() {
    var i;
    const t = this.itemOrder[0];
    return t ? ((i = this.animators.get(t)) == null ? void 0 : i.animator) ?? null : null;
  }
  forward() {
    var t, i;
    this.stop(), this.direction = "forward", this.completedCount = 0, this.completedAnimators.clear(), this.resetStartedFlags(), (i = (t = this.config).onStart) == null || i.call(t), this.startScheduler();
  }
  backward() {
    var t, i;
    this.stop(), this.direction = "backward", this.completedCount = 0, this.completedAnimators.clear(), this.resetStartedFlags(), (i = (t = this.config).onStart) == null || i.call(t), this.startScheduler();
  }
  resetStartedFlags() {
    this.animators.forEach((t) => {
      t.started = !1;
    });
  }
  stop() {
    this.stopScheduler(), this.animators.forEach((t) => {
      t.animator.stop(), t.started = !1;
    });
  }
  reverse() {
    this.direction = this.direction === "forward" ? "backward" : "forward", this.animators.forEach((t) => {
      if (!t.started)
        return;
      const i = t.animator.getCurrentState();
      if (i.position === i.to) {
        const o = new G({
          from: this.to,
          to: this.from,
          physics: t.item.physics,
          tick: t.item.tick,
          css: t.item.css,
          onComplete: () => this.onAnimatorComplete(t.id),
          onStart: t.item.onStart
        });
        o.setValue(this.to), o.setVelocity(0), o.syncState(), o.forward(), t.animator = o;
      } else
        t.animator.reverse();
    });
  }
  getCurrentState() {
    const t = this.getFirstAnimator();
    return t ? t.getCurrentState() : {
      position: 0,
      velocity: 0,
      from: 0,
      to: 1
    };
  }
  /**
   * Get current position value from first animator
   */
  getCurrentValue() {
    const t = this.getFirstAnimator();
    return (t == null ? void 0 : t.getCurrentValue()) ?? 0;
  }
  /**
   * Get current velocity from first animator
   */
  getVelocity() {
    const t = this.getFirstAnimator();
    return (t == null ? void 0 : t.getVelocity()) ?? 0;
  }
  /**
   * Set position value on first animator
   * TODO: Support per-spring control
   */
  setValue(t) {
    const i = this.getFirstAnimator();
    i == null || i.setValue(t);
  }
  /**
   * Set velocity on first animator
   * TODO: Support per-spring control
   */
  setVelocity(t) {
    const i = this.getFirstAnimator();
    i == null || i.setVelocity(t);
  }
  /**
   * Check if any animation is currently running
   */
  getIsAnimating() {
    for (const t of this.animators.values())
      if (t.animator.getIsAnimating())
        return !0;
    return !1;
  }
  /**
   * Create MultiAnimator from existing state
   * Uses first animator's state for initialization
   */
  static fromState(t, i) {
    return new D({ ...i, state: t });
  }
}
const Q = Symbol.for("TRANSITION_STRATEGY"), ht = (e) => ({
  runIn: async (n) => {
    const { currentAnimation: t } = e;
    if (t && t.direction === "out") {
      const r = t.controller.getCurrentValue(), o = t.controller.getVelocity();
      if (t.controller.stop(), n.out)
        return {
          config: await n.out,
          state: { position: r, velocity: o },
          from: 1,
          to: 0,
          direction: "backward"
          // Will actually go 0→1 (backward means toward 'from')
        };
    }
    return n.in ? {
      config: await n.in,
      state: {
        position: 0,
        velocity: 0
      },
      from: 0,
      to: 1,
      direction: "forward"
    } : {
      state: {
        position: 0,
        velocity: 0
      },
      from: 0,
      to: 1,
      direction: "forward"
    };
  },
  runOut: async (n) => {
    const { currentAnimation: t } = e;
    if (t && t.direction === "in") {
      const r = t.controller.getCurrentValue(), o = t.controller.getVelocity();
      if (t.controller.stop(), n.in)
        return {
          config: await n.in,
          state: { position: r, velocity: o },
          from: 0,
          to: 1,
          direction: "backward"
          // Will actually go 1→0 (backward means toward 'from')
        };
    }
    return n.out ? {
      config: await n.out,
      state: {
        position: 1,
        velocity: 0
      },
      from: 1,
      to: 0,
      direction: "forward"
    } : {
      state: {
        position: 1,
        velocity: 0
      },
      from: 1,
      to: 0,
      direction: "forward"
    };
  }
}), gt = () => ({
  runIn: async (e) => e.in ? {
    config: await e.in,
    state: {
      position: 0,
      velocity: 0
    },
    from: 0,
    to: 1,
    direction: "forward"
  } : {
    state: {
      position: 0,
      velocity: 0
    },
    from: 0,
    to: 1,
    direction: "forward"
  },
  runOut: async (e) => e.out ? {
    config: await e.out,
    state: {
      position: 1,
      velocity: 0
    },
    from: 1,
    to: 0,
    direction: "forward"
  } : {
    state: {
      position: 1,
      velocity: 0
    },
    from: 1,
    to: 0,
    direction: "forward"
  }
}), Z = "data-ssgoi-scope", tt = "data-ssgoi-scope-ready";
function Mt() {
  return (e) => {
    if (e)
      return e.setAttribute(Z, ""), queueMicrotask(() => {
        queueMicrotask(() => {
          e.setAttribute(tt, "true");
        });
      }), () => {
      };
  };
}
function pt(e) {
  return e.hasAttribute(tt);
}
function wt(e) {
  return e.closest(`[${Z}]`);
}
const yt = (e) => {
  let n = e.parentElement;
  for (; n && n !== document.body; ) {
    const t = window.getComputedStyle(n), i = t.overflow + t.overflowY + t.overflowX;
    if (i.includes("auto") || i.includes("scroll") || n.scrollHeight > n.clientHeight || n.scrollWidth > n.clientWidth)
      return n;
    n = n.parentElement;
  }
  return document.documentElement;
}, vt = (e) => {
  let n = e.parentElement;
  for (; n && n !== document.body; ) {
    const t = window.getComputedStyle(n).position;
    if (t === "relative" || t === "absolute" || t === "fixed" || t === "sticky")
      return n;
    n = n.parentElement;
  }
  return document.body;
};
function $(e) {
  return new Promise((n) => {
    requestAnimationFrame(() => {
      requestAnimationFrame(() => {
        n();
      });
    });
  });
}
function St(e, n) {
  var E;
  let t = null, i = null, r = null, o = null;
  const s = {
    get currentAnimation() {
      return t;
    }
  }, c = ((E = n == null ? void 0 : n.strategy) == null ? void 0 : E.call(n, s)) || ht(s), a = async (g) => {
    var S, b;
    i && (i.remove(), i = null);
    const m = e(), h = m.in && m.in(g), l = !(n != null && n.strategy) && m.out ? m.out(g) : void 0;
    if (!h)
      return;
    const T = {
      in: M(h),
      out: l ? M(l) : void 0
    }, p = await c.runIn(T);
    if (!p.config) {
      t && (t.direction = "in");
      return;
    }
    const u = p.config;
    (S = u.prepare) == null || S.call(u);
    const v = j(
      {
        ...u,
        onEnd: () => {
          var k;
          t = null, (k = u.onEnd) == null || k.call(u);
        }
      },
      g
    );
    u.wait && await u.wait();
    const w = D.fromState(p.state, {
      config: v,
      from: p.from,
      to: p.to
    });
    t = { controller: w, direction: "in" }, (b = u.onReady) == null || b.call(u), await $(), p.direction === "forward" ? w.forward() : w.backward();
  }, f = async (g) => {
    var b, k;
    i = g;
    const m = e(), h = !(n != null && n.strategy) && m.in ? m.in(g) : void 0, l = m.out && m.out(g);
    if (!l)
      return;
    const T = {
      in: h ? M(h) : void 0,
      out: M(l)
    }, p = await c.runOut(T);
    if (!p.config) {
      t && (t.direction = "out"), i && (i.remove(), i = null);
      return;
    }
    const u = p.config;
    S(), (b = u.prepare) == null || b.call(u);
    const v = j(
      {
        ...u,
        onEnd: () => {
          var z, V;
          (z = u.onEnd) == null || z.call(u), i && (i.remove(), i = null), t = null, (V = n == null ? void 0 : n.onCleanupEnd) == null || V.call(n);
        }
      },
      g
    );
    u.wait && await u.wait();
    const w = D.fromState(p.state, {
      config: v,
      from: p.from,
      to: p.to
    });
    t = { controller: w, direction: "out" }, (k = u.onReady) == null || k.call(u), await $(), p.direction === "forward" ? w.forward() : w.backward();
    function S() {
      !r || !i || (o && r.contains(o) ? r.insertBefore(i, o) : r.appendChild(i));
    }
  };
  let y = null, C = !1;
  const A = (g) => () => {
    if (C) return;
    C = !0;
    const m = g.cloneNode(!0);
    y ? queueMicrotask(() => {
      document.contains(y) && f(m);
    }) : f(m);
  };
  return (g) => {
    if (g)
      return requestAnimationFrame(() => {
        r = g.parentElement, o = g.nextElementSibling;
      }), C = !1, (n == null ? void 0 : n.scope) === "local" ? queueMicrotask(() => {
        y = wt(g), !(y && !pt(y)) && a(g);
      }) : a(g), A(g);
  };
}
const P = /* @__PURE__ */ new Map();
let R = null, U = !1;
function et(e) {
  if (e instanceof HTMLElement && P.has(e)) {
    const t = P.get(e);
    P.delete(e), queueMicrotask(() => {
      t();
    });
  }
  const n = e.childNodes;
  for (let t = 0; t < n.length; t++) {
    const i = n[t];
    i && et(i);
  }
}
function Ct() {
  U || typeof document > "u" || (U = !0, R = new MutationObserver((e) => {
    for (const n of e) {
      const t = n.removedNodes;
      for (let i = 0; i < t.length; i++) {
        const r = t[i];
        r && et(r);
      }
    }
  }), document.body ? R.observe(document.body, {
    childList: !0,
    subtree: !0
  }) : document.addEventListener(
    "DOMContentLoaded",
    () => {
      R == null || R.observe(document.body, {
        childList: !0,
        subtree: !0
      });
    },
    { once: !0 }
  ));
}
function At(e, n) {
  Ct(), P.set(e, n);
}
function Ot(e) {
  return P.has(e);
}
function Lt() {
  return P.size;
}
function Dt(e) {
  const n = P.get(e);
  return n ? (P.delete(e), n(), !0) : !1;
}
function zt() {
  P.clear(), R == null || R.disconnect(), R = null, U = !1;
}
const L = /* @__PURE__ */ new Map(), X = /* @__PURE__ */ new Map(), Y = /* @__PURE__ */ new Map();
function Tt(e, n, t) {
  L.set(e, n);
  let i = X.get(e);
  if (i)
    return i;
  const r = n;
  return i = St(
    () => {
      const o = L.get(e);
      return o || (console.warn(`Transition "${String(e)}" not found`), {});
    },
    {
      strategy: t == null ? void 0 : t.strategy,
      scope: t == null ? void 0 : t.scope,
      onCleanupEnd: () => {
        L.get(e) === r && Et(e);
      }
    }
  ), X.set(e, i), i;
}
function Et(e) {
  L.delete(e), X.delete(e), Y.delete(e);
}
function qt(e, n = "manual") {
  const t = Tt(
    e.key,
    {
      in: e.in,
      out: e.out
    },
    {
      strategy: e[Q],
      scope: e.scope
    }
  );
  if (n === "manual")
    return t;
  let i = Y.get(e.key);
  return i || (i = (r) => {
    if (r) {
      const o = t(r);
      o && At(r, o);
    }
  }, Y.set(e.key, i), i);
}
function bt(e) {
  const n = e.filter((t) => t.symmetric).map((t) => ({
    from: t.to,
    to: t.from,
    transition: t.transition
  }));
  return [...e, ...n];
}
function Rt(e) {
  let n = !1, t = 0, i = 0, r = !1;
  const o = 50, s = (g) => {
    if (!e) return;
    const m = g.touches[0];
    m && (r = m.clientX < o, r && (t = m.clientX, i = m.clientY));
  }, c = (g) => {
    if (!e || !r) return;
    const m = g.touches[0];
    if (!m) return;
    const h = m.clientX - t, l = m.clientY - i;
    h > window.outerWidth / 2 - 5 && Math.abs(h) > Math.abs(l) ? n = !0 : n = !1;
  }, a = () => {
    e && (r = !1, t = 0, i = 0);
  }, f = () => n, y = () => {
    typeof window > "u" || e && (window.addEventListener("touchstart", s, { passive: !0 }), window.addEventListener("touchmove", c, { passive: !0 }), window.addEventListener("touchend", a, { passive: !0 }));
  }, C = () => {
    typeof window > "u" || (window.removeEventListener("touchstart", s), window.removeEventListener("touchmove", c), window.removeEventListener("touchend", a));
  };
  let A = 0;
  return {
    initialize: y,
    cleanup: C,
    isSwipePending: f,
    resetSwipeDetection: () => {
      requestAnimationFrame(() => {
        A++, A > 1 && (A = 0, n = !1);
      });
    }
  };
}
function I(e, n) {
  if (n === "*")
    return !0;
  if (n.endsWith("/*")) {
    const t = n.slice(0, -1);
    return e.startsWith(t) && e.length > t.length;
  }
  return e === n;
}
function K(e, n, t) {
  for (const i of t)
    if (I(e, i.from) && I(n, i.to))
      return i.transition;
  for (const i of t)
    if ((i.from === "*" || I(e, i.from)) && (i.to === "*" || I(n, i.to)))
      return i.transition;
  return null;
}
function Pt(e = {}) {
  const { preserveScroll: n = !1, resetPatterns: t = [] } = e, i = (h) => t.some((l) => I(h, l));
  let r = null, o = null;
  const s = /* @__PURE__ */ new Map();
  let c = null, a = !1;
  const f = () => {
    r && c && !a && s.set(c, {
      x: r.scrollLeft,
      y: r.scrollTop
    });
  }, y = (h) => {
    if (!r || !n) return;
    if (i(h)) {
      r.scrollTo({ top: 0, left: 0 });
      return;
    }
    const l = s.get(h);
    if (!l) {
      r.scrollTo({ top: 0, left: 0 });
      return;
    }
    const T = 10;
    let p = 0;
    const u = () => {
      if (!r) return;
      r.scrollTo({
        top: l.y,
        left: l.x
      });
      const v = r.scrollTop, w = r.scrollLeft;
      !(Math.abs(v - l.y) < 1 && Math.abs(w - l.x) < 1) && p < T && (p++, requestAnimationFrame(u));
    };
    requestAnimationFrame(u);
  };
  return {
    initializeContext: (h, l) => {
      a = !0, o = h, r || (r = yt(h), (r === document.documentElement ? window : r).addEventListener("scroll", f, {
        passive: !0
      })), c = l, y(l);
      const T = 10;
      let p = 0;
      const u = () => {
        p++, p >= T ? a = !1 : requestAnimationFrame(u);
      };
      requestAnimationFrame(u);
    },
    calculateScrollOffset: (h, l) => {
      const T = h && s.has(h) ? s.get(h) : { x: 0, y: 0 }, p = l && i(l);
      p && s.set(l, { x: 0, y: 0 });
      const u = p ? { x: 0, y: 0 } : l && s.has(l) ? s.get(l) : { x: 0, y: 0 };
      return {
        x: -u.x + T.x,
        y: -u.y + T.y
      };
    },
    getScrollContainer: () => r,
    getPositionedParentElement: () => o ? vt(o) : document.body,
    getScrollPosition: (h) => h && s.has(h) ? s.get(h) : { x: 0, y: 0 }
  };
}
function kt() {
  let e = null;
  function n() {
    if (e != null && e.from && (e != null && e.to) && (e != null && e.outResolve) && (e != null && e.inResolve)) {
      const t = { from: e.from, to: e.to };
      e.outResolve(t), e.inResolve(t), e = null;
    }
  }
  return {
    trigger(t, i) {
      e || (e = {}), i === "out" ? e.from = t : e.to = t;
    },
    get(t) {
      return t === "in" && (!e || !e.from) ? Promise.resolve(null) : new Promise((i) => {
        e || (e = {}), t === "out" ? e.outResolve = i : e.inResolve = i, n();
      });
    }
  };
}
function Vt() {
  let e = null;
  function n() {
    var i, r;
    e && ((i = e.outResolve) == null || i.call(e, null), (r = e.inResolve) == null || r.call(e, null), e = null);
  }
  function t() {
    if (e != null && e.from && (e != null && e.to) && (e != null && e.outResolve) && (e != null && e.inResolve)) {
      const i = { from: e.from, to: e.to };
      e.outResolve(i), e.inResolve(i), e = null;
    }
  }
  return {
    trigger(i, r) {
      e && (r === "out" && e.from && e.from !== i || r === "in" && e.to && e.to !== i) && n(), e || (e = {}), r === "out" ? e.from = i : e.to = i;
    },
    get(i) {
      return new Promise((r) => {
        e || (e = {}), i === "out" ? e.outResolve = r : e.inResolve = r, t();
      });
    }
  };
}
function _t(e, n) {
  const {
    transitions: t = [],
    defaultTransition: i,
    middleware: r = (v, w) => ({ from: v, to: w }),
    // Identity function as default
    skipOnIosSwipe: o = !0,
    // Default to true - skip animations on iOS swipe
    experimentalPreserveScroll: s = !1,
    // Default to false - manual scroll management
    scrollResetPatterns: c = []
    // Default to empty array - no routes reset scroll
  } = e, { outFirst: a = !0, createNavigationDetector: f } = n || {}, y = (f == null ? void 0 : f()) ?? (a ? kt() : Vt()), C = bt(t), {
    initializeContext: A,
    calculateScrollOffset: E,
    getScrollContainer: g,
    getPositionedParentElement: m,
    getScrollPosition: h
  } = Pt({
    preserveScroll: s,
    resetPatterns: c
  }), l = Rt(o);
  l.initialize();
  const T = async (v, w) => {
    if (l.isSwipePending())
      return l.resetSwipeDetection(), w === "in" ? async (F) => ({
        onReady: () => {
          F.style.visibility = "visible";
        }
      }) : () => ({});
    y.trigger(v, w);
    const S = await y.get(w);
    if (!S) return () => ({});
    const { from: b, to: k } = r(
      S.from,
      S.to
    ), V = K(
      b,
      k,
      C
    ) || i;
    if (!V) return () => ({});
    const H = E(S.from, S.to);
    if (w === "out") {
      const F = {
        scrollOffset: H,
        scroll: h(S.from),
        get scrollingElement() {
          return g() || document.documentElement;
        },
        get positionedParent() {
          return m();
        }
      };
      return (x) => V.out(x, F);
    } else {
      const F = {
        scrollOffset: H,
        get scroll() {
          return h(S.to);
        },
        get scrollingElement() {
          return g() || document.documentElement;
        },
        get positionedParent() {
          return m();
        }
      };
      return async (x) => {
        const q = await Promise.resolve(V.in(x, F)), _ = q.onReady;
        return q.onReady = () => {
          x.style.visibility = "visible", _ == null || _();
        }, q;
      };
    }
  }, p = (v) => ({
    key: v,
    in: async (w) => (A(w, v), (await T(v, "in"))(w)),
    out: async (w) => (await T(v, "out"))(w),
    // Add page transition strategy for page-level transitions
    [Q]: gt
  }), u = (v, w) => {
    if (l.isSwipePending())
      return !1;
    const { from: S, to: b } = r(v, w);
    return !!(K(
      S,
      b,
      C
    ) || i);
  };
  return f ? {
    getTransition: p,
    hasMatchingTransition: u
  } : p;
}
export {
  Z as SCOPE_ATTR,
  tt as SCOPE_READY_ATTR,
  Q as TRANSITION_STRATEGY,
  Vt as createAnyOrderDetector,
  ht as createDefaultStrategy,
  kt as createOutFirstDetector,
  gt as createPageTransitionStrategy,
  _t as createSggoiTransitionContext,
  Mt as createTransitionScope,
  wt as findScope,
  Lt as getWatchedCount,
  Xt as isCssAnimation,
  Yt as isMultiAnimation,
  pt as isScopeReady,
  Ht as isSingleAnimation,
  jt as isTickAnimation,
  Ot as isWatched,
  j as normalizeSchedule,
  M as normalizeToMultiAnimation,
  zt as resetObserver,
  qt as transition,
  Dt as triggerUnmount,
  At as watchUnmount
};
